src/core/thread/mesh_forwarder.cpp

```diff
@@ -1120,6 +1120,12 @@ ThreadError MeshForwarder::HandleFrameRequest(Mac::Frame &aFrame)
     }
 
 exit:
+
+    if (error != kThreadError_None)
+    {
+        mSendBusy = false;
```

Comment:
I do not think you should clear this flag.  The MAC layer will always signal `HandleTransmitDone()` even if we abort here.